include/circt/Dialect/FIRRTL/Passes.td

```diff
@@ -210,4 +210,12 @@ def GrandCentralTaps : Pass<"firrtl-grand-central-taps", "firrtl::CircuitOp"> {
   let dependentDialects = ["sv::SVDialect"];
 }
 
+def CheckCombCycles : Pass<"firrtl-check-comb-cycles", "firrtl::CircuitOp"> {
```

Comment:
Please make this a pass that runs on ModuleOp so it runs in parallel.

Reply:
Auh, forgot to reply on your last comment. We need to consider cross-module combinational cycles here, because a module may have combinational paths between its IOs.

In the current implementation, we encode these combinational paths into the `CombPaths` data structure and maintain a map from `FModuleOp` to `CombPaths`. When constructing the combinational graph of a module, the combinational paths of sub-instances will be inlined into the graph. Therefore, we traverse Andrew's instance graph in a post order in this pass.

Parallelization is still possible by cutting the instance graph into a tree and running on all leaf nodes in parallel, but this will still happen inside of a `CircuitOp` pass.